packages/plugin/src/theia.d.ts

```diff
@@ -5145,6 +5145,22 @@ declare module '@theia/plugin' {
          */
         export let workspaceFolders: WorkspaceFolder[] | undefined;
 
+        /**
+         * The location of the workspace file, for example:
+         *
+         * `file:///Users/name/Development/myProject.code-workspace`
+         *
+         * Depending on the workspace that is opened, the value will be:
+         *  * `undefined` when no workspace or a single folder is opened
+         *  * the path of the workspace file as `Uri` otherwise.
+         *
+         * **Note:** it is not advised to use `workspace.workspaceFile` to write
+         * configuration data into the file.
+         * 
+         * @readonly
+         */
+        export const workspaceFile: Uri | undefined;
```

Comment:
Just to confirm, why `const`?
In the case that a root is added, do we not want to update from `undefined` to `URI`?

Reply:
I think this indicates that the value is a snapshot and should not be modified by the extension, this is defined in the same way as vscode, at the same time the const does not prevent the extension host to resolve the property from a variable.